cpp/FileTransferAgent.hpp

```diff
@@ -61,9 +61,11 @@ class RetryContext
      */
     bool isRetryable(RemoteStorageRequestOutcome putStatus)
     {
-        //If putStatus is not SUCCESS and not TOKEN_EXPIRED then put is retryable
+        //If putStatus is not SUCCESS, not TOKEN_EXPIRED and not SKIP_UPLOAD_FILE then put is retryable
         bool isPutInRetryableState = ((putStatus != RemoteStorageRequestOutcome::SUCCESS) &&
-                (putStatus != RemoteStorageRequestOutcome::TOKEN_EXPIRED)) ;
```

Comment:
So, seems isPutInRetryableState is equivalent to putStatus == RemoteStorageRequestOutcome ::FAILED

Reply:
Yes it is, Just wanted to be explicit on when we will be retrying.